docs/developing.rst

```diff
@@ -742,3 +742,246 @@ Note the call to :func:`toil.job.Job.encapsulate` creates the
 :class:`toil.job.Job.EncapsulatedJob`.
 
 
+Deploying a workflow
+====================
+
+If a Toil workflow is run on a single machine, there is nothing special you
+need to do. You change into the directory containing your user script and
+invoke like any Python script::
+
+   $ cd my_project
+   $ ls
+   userScript.py …
+   $ ./userScript.py …
+
+This assumes that your script is executable and contains a *shebang*, i.e. a
+line of the form
+
+::
+
+   #!/usr/bin/python
+
+or equivalent. The script can have dependencies, as long as those are installed
+on the machine, either globally, in a user-specific location or in a
+virtualenv. In the latter case, the virtualenv must of course be active when
+you run the user script.
+
+If, however, you want to run your workflow in a distributed environment, on
+multiple worker machines, either in the cloud or on a bare-metal cluster, your
+script needs to be made available to those other machines. If your script
+imports other modules, those modules also need to be made available on the
+workers. Toil can automatically do that for you, with a little help on your
+part. We call this feature ``hot-deployment`` of a workflow.
+
+
+Hot-deployment without dependencies
+-----------------------------------
+
+Your script has no additional dependencies, it imports only modules that are
```

Comment:
Nit: it reads a bit weird to start with a declarative like this. Perhaps "If your script has no additional dependencies..." with accompanying changes downstream?